GitCommands/Git/ExecutableException.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+using JetBrains.Annotations;
+
+namespace GitCommands.Git
+{
+    public class ExecutableException : ExternalOperationException
+    {
+        public ExecutableException(ProcessStartInfo processStartInfo, [NotNull] Exception inner)
+            : base(inner)
+        {
+            ProcessStartInfo = processStartInfo;
+            if (string.IsNullOrEmpty(ProcessStartInfo.WorkingDirectory))
+            {
+                ProcessStartInfo.WorkingDirectory = Directory.GetCurrentDirectory();
+            }
```

Comment:
Why does the exception do this? The exception is merely a carrier of error information, it mustn't modify information it carries.

Reply:
> Why does the exception do this? The exception is merely a carrier of error information, it mustn't modify information it carries.

I agree in general. But in this case, it adds missing information.
Alternatively, `Executable`/`ProcessWrapper` could specify the `WorkingDirectory` always.